examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,401 @@
+# evaluate the performance of the model and its capacity to generalize but
+# rather its capacity to learn from the training data.
+
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.model_selection import train_test_split
+
+max_iter_list = [5, 50]
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, shuffle=False)
+average_week_demand = (
+    df.loc[X_test.index].groupby(["day", "period"], observed=False)["transfer"].mean()
+)
+
+colors = sns.color_palette("colorblind")
+fig, ax = plt.subplots(figsize=(10, 5))
+average_week_demand.plot(color=colors[0], label="ground truth", linewidth=2, ax=ax)
+
+for idx, max_iter in enumerate(max_iter_list):
+    hgbt = HistGradientBoostingRegressor(max_iter=max_iter)
+    hgbt.fit(X_train, y_train)
+
+    y_pred = hgbt.predict(X_test)
+    prediction_df = df.loc[X_test.index].copy()
+    prediction_df["y_pred"] = y_pred
+    average_pred = prediction_df.groupby(["day", "period"], observed=False)[
+        "y_pred"
+    ].mean()
+    average_pred.plot(
+        color=colors[idx + 1], label=f"max_iter={max_iter}", linewidth=2, ax=ax
+    )
+
+ax.set(
+    title="Predicted average energy transfer during the week",
+    xticks=[(i + 0.2) * 48 for i in range(7)],
+    xticklabels=["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend()
+
+# %%
+# With just a few iterations, HGBT models can achieve convergence (see
+# :ref:`sphx_glr_auto_examples_ensemble_plot_forest_hist_grad_boosting_comparison.py`).
+#
+# Instead of relying solely on `max_iter` to determine when to stop, the HGBT
+# implementations in scikit-learn support early stopping. With it, the model
+# uses a fraction of the training data as a validation set
+# (`validation_fraction`) and stops training if the validation score does not
+# improve (or degrades) after `n_iter_no_change` iterations up to a certain
+# `tol`.
+#
+# Notice that there is a trade-off between `learning_rate` and `max_iter`:
+# Generally, smaller learning rates require more iterations to converge to the
+# minimum loss, while larger learning rates might converge faster but are at
+# risk of overfitting.
+#
+# Indeed, a good practice is to tune the learning rate along with any other
+# hyperparameters, fit the HBGT on the training set with a large enough value
+# for `max_iter` and determine the best `max_iter` via early stopping and some
+# explicit `validation_fraction`.
+
+common_params = {
+    "max_iter": 1_000,
+    "learning_rate": 0.3,
+    "validation_fraction": 0.2,
+    "random_state": 42,
+    "scoring": "neg_root_mean_squared_error",
+}
+
+hgbt = HistGradientBoostingRegressor(early_stopping=True, **common_params)
+hgbt.fit(X_train, y_train)
+plt.plot(-hgbt.validation_score_)
+plt.xlabel("number of iterations")
+plt.ylabel("root mean squared error")
+_ = plt.title(f"Loss of hgbt with early stopping (n_iter={hgbt.n_iter_})")
```

Comment:
Could use the `_, ax = plt.subplots()` and then `ax` to make the plot.
This is recommended in the matplotlib documentation if I recall correctly.